src/vmm/src/builder.rs

```diff
@@ -970,9 +988,9 @@ pub mod tests {
     }
 
     pub(crate) fn default_kernel_cmdline() -> Cmdline {
-        let mut kernel_cmdline = kernel::cmdline::Cmdline::new(4096);
-        kernel_cmdline.insert_str(DEFAULT_KERNEL_CMDLINE).unwrap();
-        kernel_cmdline
+        let mut boot_cmdline = linux_loader::cmdline::Cmdline::new(arch::CMDLINE_MAX_SIZE);
```

Comment:
well the default is much smaller (100 chars), so why don't we allocate the actual size of it?
If not, we may be allocating 65536 unused bytes (on x86)

Reply:
we append to it so the size will actually grow, that is why we make it bigger than default value